shared/src/validation-schemas/user/user-sign-up.validation-schema.ts

```diff
@@ -2,17 +2,31 @@ import * as Joi from 'joi';
 import { getNameOf } from '~/helpers/helpers';
 import { UserSignUpRequestDto } from '~/common/types/types';
 import { UserValidationMessage } from '~/common/enums/enums';
+import {
+  EMAIL_MAX_LENGTH,
+  EMAIL_MIN_LENGTH,
+  EMAIL_PATTERN,
+  PASSWORD_MAX_LENGTH,
+  PASSWORD_MIN_LENGTH,
+} from '~/common/constants/constants';
 
 const userSignUp = Joi.object({
   [getNameOf<UserSignUpRequestDto>('email')]: Joi.string()
     .trim()
     .email({ tlds: { allow: false } })
     .required()
+    .min(EMAIL_MIN_LENGTH)
+    .max(EMAIL_MAX_LENGTH)
+    .pattern(EMAIL_PATTERN)
```

Comment:
I think `joi` may deal with email string format validation,, email regex pattern matching may be an overkill

Reply:
Totally agree, did it because it was as a request on Trello ticket from a QA team.
Fixed it.